jgrapht-core/src/test/java/org/jgrapht/alg/tour/GeometricTSPTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (C) Copyright 2019-2019, by Peter Harman and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.tour;
+
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PrimitiveIterator.OfDouble;
+import java.util.Random;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.SlowTests;
+import org.jgrapht.alg.interfaces.HamiltonianCycleAlgorithm;
+import static org.jgrapht.alg.tour.TwoApproxMetricTSPTest.assertHamiltonian;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.builder.GraphBuilder;
+import org.jgrapht.graph.builder.GraphTypeBuilder;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Tests of Travelling Salesman Problem algorithms based on a random set of 2D
+ * points, with graphs of increasing size
+ *
+ * @author Peter Harman
+ */
+@Category(SlowTests.class)
+@RunWith(Parameterized.class)
+public class GeometricTSPTest {
+
+    private static final Logger LOG = Logger.getLogger(GeometricTSPTest.class.getName());
+
+    private static final OfDouble RNG = new Random().doubles(0.0, 100.0).iterator();
+    private final Graph<Point2D, DefaultWeightedEdge> graph;
+
+    public GeometricTSPTest(Graph<Point2D, DefaultWeightedEdge> graph) {
+        this.graph = graph;
+    }
+
+    @Parameterized.Parameters
+    public static Object[][] graphs() {
+        List<Object[]> graphs = new ArrayList<>();
+        for (int i = 0; i < 4; i++) {
+            graphs.add(new Object[]{generate((int) Math.pow(10, i))});
+        }
+        return graphs.toArray(new Object[0][]);
+    }
+
+    static Graph<Point2D, DefaultWeightedEdge> generate(int n) {
+        Point2D[] points = new Point2D[n];
+        for (int i = 0; i < n; i++) {
+            points[i] = new Point2D.Double(RNG.next(), RNG.next());
```

Comment:
Have you considered `org.apache.commons.math3.geometry.euclidean.twod.Vector2D` instead of `java.awt.geom.Point2D`?
I have to admint it is only an semantical improvement because it has the same internal fields and provides the methods you use, but `Point2D` is provided by the GUI-toolkit AWT it is more meaningfull to use `Point2D` in conjunction with GUI tasks. `Vector2D` on the other hand is part of an general purpose math library which could be slightly more suitable here.

Reply:
Done